### PR TITLE
[IOTDB-4482] IoTDB distribution: using id to delete nodes

### PR DESCRIPTION
**Pull request by Caideyipi**
### 1
Input id instead of IP and port for node deleting;
### 2
Changed some logs;
### 3
Simplified some codes

Test images:
Before deletion:
image
remove-confignode:
image
image
remove-datanode:
image
image
After deletion:
image

<hr>

This PR has:
- [x] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Review by wangchao316 — CHANGES_REQUESTED** [review]
hi,I don't think it's a good change. We can easily get the ip and port from the configuration file.

File: confignode/src/main/java/org/apache/iotdb/confignode/service/ConfigNodeCommandLine.java [hunk 1]
```diff
@@ -97,18 +95,17 @@ protected int run(String[] args) {
 
   private void doRemoveNode(String[] args) throws IOException {
     LOGGER.info("Starting to remove {}...", ConfigNodeConstant.GLOBAL_NAME);
-    if (args.length != 3) {
-      LOGGER.info("Usage: -r <internal_address>:<internal_port>");
+    if (args.length != 2) {
+      LOGGER.info("Usage: <Node-id>");
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/service/ConfigNodeCommandLine.java:99]:
```suggestion
      LOGGER.info("Usage: -r <Node-id>");
```

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java [hunk 2]
```diff
@@ -104,52 +98,27 @@ protected int run(String[] args) throws Exception {
...
-      List<String> onlineDataNodeIps =
-          nodeIdToNodeConfiguration.values().stream()
-              .map(TDataNodeConfiguration::getLocation)
-              .map(TDataNodeLocation::getInternalEndPoint)
-              .map(TEndPoint::getIp)
-              .collect(Collectors.toList());
-      IoTDBStopCheck.getInstance().checkIpInCluster(removedDataNodeIps, onlineDataNodeIps);
+    logger.info("Start to remove DataNode from cluster");
+    if (args.length != 2) {
+      // logger.info("Usage: <node-ids>, split with ':'");
```
Comment (CRZbulabula) [inline server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:107]: Don't leave commented codes~

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java [hunk 3]
```diff
@@ -104,52 +98,27 @@ protected int run(String[] args) throws Exception {
...
+      logger.info("Usage: <node-id>");
```
Comment (CRZbulabula) [inline server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:108]:
```suggestion
      logger.info("Usage: -r <node-id>");
```

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java [hunk 4]
```diff
@@ -104,52 +98,27 @@ protected int run(String[] args) throws Exception {
...
+      throw new BadNodeUrlException("No DataNode to remove");
     }
-    logger.info(
-        "there has data nodes location will be removed. size is: {}, detail: {}",
-        dataNodeLocations.size(),
-        dataNodeLocations);
+    //    logger.info(
+    //        "There are DataNodes to be removed. size is: {}, detail: {}",
+    //        dataNodeLocations.size(),
+    //        dataNodeLocations);
```
Comment (CRZbulabula) [inline server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:120]: Don't leave commented codes~

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java [hunk 5]
```diff
@@ -159,51 +128,52 @@ private void removeNodesFromCluster(String[] args)
...
     List<TDataNodeLocation> dataNodeLocations = new ArrayList<>();
-    if (endPorts == null || endPorts.trim().isEmpty()) {
+    if (ids == null || ids.trim().isEmpty()) {
       return dataNodeLocations;
     }
 
-    List<TEndPoint> endPoints = NodeUrlUtils.parseTEndPointUrls(endPorts);
+    // Currently support only single id, delete this line when supports multi ones
+    if (ids.split(":").length > 1) {
+      logger.info("Incorrect id format, usage: <node-id>");
```
Comment (CRZbulabula) [inline server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:152]:
```suggestion
      logger.info("Incorrect id format, usage: -r <node-id>");
```

**Review by CRZbulabula — CHANGES_REQUESTED** [review]
PTAL. By the way, it's better to keep the old ip:port interface while add the id interface as a new function.

File: confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeRemoveCheck.java [hunk 6]
```diff
@@ -50,19 +51,34 @@ public ConfigNodeRemoveCheck() {
...
-              .findFirst()
-              .orElse(null);
+      try {
+        int id = Integer.parseInt(args);
+        nodeLocation =
+            getConfigNodeList().stream()
+                .filter(e -> e.getConfigNodeId() == id)
+                .findFirst()
+                .orElse(null);
+      } catch (NumberFormatException e1) {
```
Comment (Beyyes) [inline confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeRemoveCheck.java:69]: Using `isNumeric` method to judge but not using `NumberFormatException `

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java [hunk 7]
```diff
@@ -159,51 +125,66 @@ private void removeNodesFromCluster(String[] args)
...
+            client.getDataNodeConfiguration(-1).getDataNodeConfigurationMap().values().stream()
+                .map(TDataNodeConfiguration::getLocation)
+                .filter(location -> endPoints.contains(location.getClientRpcEndPoint()))
+                .collect(Collectors.toList());
+      } catch (TException e) {
+        logger.error("get data node locations failed", e);
+      }
+    } catch (BadNodeUrlException e) {
+      try (ConfigNodeClient client = new ConfigNodeClient()) {
+        for (String id : args.split(",")) {
```
Comment (Beyyes) [inline server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:167]: Now we only support removing one datanode id, if it's necessary to using `args.split(",")`

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java [hunk 8]
```diff
@@ -104,52 +100,22 @@ protected int run(String[] args) throws Exception {
...
-      List<String> onlineDataNodeIps =
-          nodeIdToNodeConfiguration.values().stream()
-              .map(TDataNodeConfiguration::getLocation)
-              .map(TDataNodeLocation::getInternalEndPoint)
-              .map(TEndPoint::getIp)
-              .collect(Collectors.toList());
-      IoTDBStopCheck.getInstance().checkIpInCluster(removedDataNodeIps, onlineDataNodeIps);
+    logger.info("Start to remove DataNode from cluster");
+    if (args.length != 2) {
+      logger.info("Usage: <node-id>/<ip>:<rpc-port>");
```
Comment (Beyyes) [inline server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:109]: Also update the user doc

File: confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeRemoveCheck.java (hunk 6 above)
Comment (Caideyipi) [reply confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeRemoveCheck.java:69]: Solved

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java (hunk 7 above)
Comment (Caideyipi) [reply server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:169]: Reserving multi-node deletion logic is suggested by boss Qiao~

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java (hunk 8 above)
Comment (Caideyipi) [reply server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:111]: Updated

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java (hunk 5 above)
Comment (Caideyipi) [reply server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java]: I deem the '-r' somehow redundant, so I removed it in the command

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java (hunk 4 above)
Comment (Caideyipi) [reply server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java]: Deleted

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java (hunk 3 above)
Comment (Caideyipi) [reply server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java]: I deem the '-r' somehow redundant, so I removed it in the command

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java (hunk 2 above)
Comment (Caideyipi) [reply server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java]: Deleted

File: confignode/src/main/java/org/apache/iotdb/confignode/service/ConfigNodeCommandLine.java (hunk 1 above)
Comment (Caideyipi) [reply confignode/src/main/java/org/apache/iotdb/confignode/service/ConfigNodeCommandLine.java]: I deem the '-r' somehow redundant, so I removed it in the command

**Comment by Caideyipi** [comment]
> hi,I don't think it's a good change. We can easily get the ip and port from the configuration file.
IP and port is re-supported in the current version

File: server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java [hunk 9]
```diff
@@ -159,51 +127,67 @@ private void removeNodesFromCluster(String[] args)
...
+        logger.error("Get data node locations failed", e);
+      }
+    } catch (BadNodeUrlException e) {
+      try (ConfigNodeClient client = new ConfigNodeClient()) {
+        for (String id : args.split(",")) {
+          if (!isNumeric(id)) {
+            logger.warn("Incorrect id format {}, skipped...", id);
+            continue;
+          }
+          List<TDataNodeLocation> NodeLocationResult =
```
Comment (Beyyes) [inline server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java:174]:
```suggestion
          List<TDataNodeLocation> nodeLocationResult =
```

**Review by CRZbulabula — APPROVED** [review]
LGTM～

**Review by Beyyes — APPROVED** [review]
LGTM

**Pull request merged**